matscipy/gamma_surface.py

```diff
@@ -0,0 +1,699 @@
+            Generated gamma surface images (populated after self.generate_images is called)
+        '''
+
+        self.images = []
+        self.calc = None
+        self.nx = 0
+        self.ny = 0
+        self.x_disp = 0
+        self.y_disp = 0
+        self.surface_area = 0
+        self.offset = 0
+        self.crystalstructure = None
+        self.ylims = [0, 1]
+        self.Es = None
+
+        axes = None
+
+        # Check if surface_direction is some kind of dislocation
+        disloc = False
+        if inspect.isclass(surface_direction):
+            if issubclass(surface_direction, CubicCrystalDislocation):
+                # Passed a class
+                disloc = True
+                dissociated = issubclass(surface_direction, CubicCrystalDissociatedDislocation)
+        elif isinstance(surface_direction, CubicCrystalDislocation):
+            # Passed an instance
+            disloc = True
+            dissociated = isinstance(surface_direction, CubicCrystalDissociatedDislocation)
+        
+        if disloc:
+            # surface_direction was some kind of CubicCrystalDislocation
+            if dissociated:
+                disloc = surface_direction.left_dislocation
+            else:
+                disloc = surface_direction
+            # Dislocation object was found
+            axes = disloc.axes.copy()
+            self.offset = -disloc.unit_cell_core_position_dimensionless[1]
+            crystalstructure = disloc.crystalstructure
+
+            self.ylims = [0, disloc.glide_distance_dimensionless]
+
+            self.surf_directions = {
+                "x": axes[2, :],
+                "y": axes[0, :],
+                "z": axes[1, :]
+            }
+            ax = axes.copy()
+            ax[0, :] = self.surf_directions["x"]
+            ax[1, :] = self.surf_directions["y"]
+            ax[2, :] = self.surf_directions["z"]
+        else:
+            # surface_direction is a vector for the basis
+            z, y, x = complete_basis(surface_direction, glide_direction)
+
+            # z, y, x -> x, y, z swap means chirality is wrong
+            if glide_direction is None:
+                y, x = x, y
+            else:
+                x = - x
+
+            self.surf_directions = {
+                "x": np.array(x),
+                "y": np.array(y),
+                "z": np.array(z)
+                }
+            ax = np.array([x, y, z])
+        
+        alat, self.cut_at = validate_cubic_cell(a, axes=ax, crystalstructure=crystalstructure, symbol=symbol)
+        self.offset *= alat
+
+    def _vec_to_miller(self, vec):
+        '''
+        Converts np.array vec to string miller notation
+
+        vec: np.array
+        array to convert to string miller notation
+        e.g. (100) or (11-2)
+        '''
+        return "(" + "".join(vec.astype(int).astype(str)) + ")"
+
+    def _gen_compressed_images(self, base_struct, nx, ny, x_points, y_points, vacuum=0.0):
```

Comment:
compressed -> cell, uncompressed -> positions throughout

when need a bool to choose between them use `cell_move=True/False`